solvers/cd.py

```diff
@@ -0,0 +1,135 @@
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+with safe_import_context() as import_ctx:
+    import math
+    import numpy as np
+    from scipy import sparse
+    from numba import njit
+
+
+if import_ctx.failed_import:
+
+    def njit(f):  # noqa: F811
+        return f
+
+
+@njit
+def _newton_step_size(X, exp_yXw, j, lmbd):
+    hess_jj = 0.
+    for i in range(len(X)):
+        hess_jj += X[i, j]**2 * exp_yXw[i] / (1 + exp_yXw[i]**2)
```

Comment:
I believe the denominator should be `(1 + exp_yXw[i]) ** 2`

It's the second derivative of `ln(1 + exp((u))`, first derivative is sigmoid(u) and second is `sigmoid(u) * (1 - sigmoid(u))`